app/src/main/java/com/tomaszpolanski/androidsandbox/ActivityIntent.kt

```diff
@@ -0,0 +1,34 @@
+package com.tomaszpolanski.androidsandbox
+
+import android.os.Bundle
+import android.os.Parcelable
+import android.support.v7.app.AppCompatActivity
+
+abstract class ActivityIntent<S : Parcelable, E> : AppCompatActivity() {
+
+    private val stateKey = javaClass.name
+    private var lastState: S? = null
+    protected lateinit var viewModel: ViewModelBase<S, E>
+
+    protected open fun render(state: S) {
```

Comment:
would it be better to rename this method to `updateState` and have it call an abstract method `render`? that way you don't have to worry about calling `super`

Reply:
Sounds good :)